posts/month-in-mathlib-may-2024.md

```diff
@@ -0,0 +1,927 @@
+    * [PR #12526](https://github.com/leanprover-community/mathlib4/pull/12526) defines locally linear graphs, [PR #12570](https://github.com/leanprover-community/mathlib4/pull/12570) constructs such graphs from a set of specified triangles respecting some conditions, [PR #12523](https://github.com/leanprover-community/mathlib4/pull/12523) uses that construction to deduce the Triangle removal lemma from the Regularity lemma.
+    * [PR #12701](https://github.com/leanprover-community/mathlib4/pull/12701) redefines sets without arithmetic progressions of length 3 (aka 3AP-free sets) so that they behave correctly in characteristic two. [PR #12546](https://github.com/leanprover-community/mathlib4/pull/12546) refactors Freiman homomorphisms and isomorphisms from a bundled design to unbundled predicates. This makes them much easier to use. [PR #12551](https://github.com/leanprover-community/mathlib4/pull/12551) then proves the no wrap-around principle stating that additive structure in sets is independent of the ambient group so long as torsion is much bigger than the diameter of the sets.
+    * Building up on thoses two series of PRs, [PR #13074](https://github.com/leanprover-community/mathlib4/pull/13074) defines corners and corner-free set and [PR #9000](https://github.com/leanprover-community/mathlib4/pull/9000) finally proves the Corners theorem and Roth's theorem. They respectively state that a corner-free set in `[N] × [N]` and a 3AP-free set in `[N]` have vanishingly small density as `N` goes to infinity.
+
+    [APAP](https://github.com/YaelDillies/LeanAPAP) already contains the stronger result that the density goes to zero as `1/log log N`, and will soon prove the state of the art result of `exp(-(log N)^1/9)`.
+  * Mathlib finally knows about Hamiltonian paths and cycles thanks to a team effort started at Lean for the Curious Mathematician 2023 in Düsseldorf by Rishi Mehta and Linus Sommer under the supervision of Bhavik Mehta, and recently continued by Lode Vermeulen in [PR #7102](https://github.com/leanprover-community/mathlib4/pull/7102).
+
+* Condensed mathematics.
+* [PR #11586](https://github.com/leanprover-community/mathlib4/pull/11586) :: feat(Condensed): light condensed objects
+* [PR #12870](https://github.com/leanprover-community/mathlib4/pull/12870) :: feat(Condensed): prove the explicit sheaf condition for a general target category
+* [PR #13501](https://github.com/leanprover-community/mathlib4/pull/13501) :: feat(Condensed): light condensed modules
+* [PR #13503](https://github.com/leanprover-community/mathlib4/pull/13503) :: feat(Condensed): explicit sheaf condition for light condensed sets
+* [PR #13500](https://github.com/leanprover-community/mathlib4/pull/13500) :: feat(Condensed): discrete light condensed objects
+* [PR #13502](https://github.com/leanprover-community/mathlib4/pull/13502) :: feat(Condensed): functor from light profinite sets to light condensed sets
```

Comment:
```suggestion
* Condensed mathematics.
  * The work towards getting the basics of condensed mathematics into mathlib started about a year ago at a workshop in Copenhagen organized by Dagur Asgeirsson. The workshop participants made great progress and proved results like the characterization of condensed sets as finite-product-preserving presheaves on `Stonean`. Since then, Dagur has been moving the material to mathlib and developing it further. In May, the definition of a light condensed object made it to mathlib (see [PR #11586](https://github.com/leanprover-community/mathlib4/pull/11586)). This is a concept introduced by Clausen and Scholze less than a year ago, and is an improvement over condensed sets in that it removes certain size-issues, as it is a sheaf over an essentially small site. Another result that landed in mathlib in May is the explicit sheaf condition ([PR #12870](https://github.com/leanprover-community/mathlib4/pull/12870)) for condensed objects in a very general target category (we had this only in certain concrete categories before).
```

Reply:
Maybe this is too long, tell me if you want me to cut something, or feel free to do it yourself

Reply:
No, looks good!

Reply:
Did you mean to commit this? The conversation is resolved but it's still the old text.

Reply:
Sorry, I have been badly multitasking all of today